mayastor/tests/nexus_rebuild.rs

```diff
@@ -226,15 +244,269 @@ async fn nexus_test_child(child: u64) {
         get_dev(child),
         RebuildState::Completed,
         std::time::Duration::from_secs(10),
-    );
+    )
+    .unwrap();
+
+    let nexus = nexus_lookup(nexus_name()).unwrap();
 
     let (s, r) = unbounded::<String>();
     std::thread::spawn(move || {
         s.send(common::compare_devices(
             &get_disk(0),
             &get_disk(child),
+            nexus.size(),
             true,
         ))
     });
     reactor_poll!(r);
 }
+
+#[test]
+// test rebuild with different combinations of sizes for src and dst children
+fn rebuild_sizes() {
+    test_ini("rebuild_sizes");
+
+    let nexus_size = 10 * 1024 * 1024; // 10MiB
+    let child_size = nexus_size + META_SIZE;
+    let mut test_cases = vec![
+        (nexus_size, child_size, child_size),
+        (nexus_size, child_size * 2, child_size),
+        (nexus_size, child_size, child_size * 2),
+        (nexus_size, child_size * 2, child_size * 2),
+    ];
+    // now for completeness sake we also the cases where the actual
+    // nexus_size will be lower due to the on-disk metadata
+    let child_size = nexus_size;
+    test_cases.extend(vec![
+        (nexus_size, child_size, child_size),
+        (nexus_size, child_size * 2, nexus_size),
```

Comment:
Was "nexus_size" meant to be used twice here?

Reply:
Done